### PR TITLE
Fix connect cell monogram truncation

### PR DESCRIPTION
**Pull request by stevemoser**
When an emoji is used in monogram label in the connect TVC sometimes the monogram becomes truncated. Turning on adjustsFontSizeToFitWidth is a quick fix but other fixes may be desired. I believe the phone and contacts apps on iOS display a default avatar image when an emoji is the first character in a contact's first or last name. An emoji might be a silly example but other unicode glyphs could become truncated as well. See the before and after images.

truncated-monogram

fixed-monogram

### CODE REVIEW

**Comment by umerkhan-apple** [comment]
@stevemoser Thanks for this PR! 

I think we could combine `adjustsFontSizeToFitWidth` along with clipping the string length to be 2 characters. This is to prevent the monogram from becoming a really small font size if a long string is provided. What do you think?

File: CareKit/Connect/OCKConnectTableViewCell.m [hunk 1]
```diff
@@ -102,7 +103,12 @@ - (void)updateView {
     } else {
         _imageView.image = nil;
         _imageView.backgroundColor = OCKSystemGrayColor();
-        _monogramLabel.text = self.contact.monogram;
+        
+        NSString *monogram = self.contact.monogram;
+        NSRange stringRange = {0, MIN([monogram length], 2)};
+        stringRange = [monogram rangeOfComposedCharacterSequencesForRange:stringRange];
+        NSString *clippedMonogram = [monogram substringWithRange:stringRange];
+        _monogramLabel.text = clippedMonogram;
```
Comment (stevemoser) [inline CareKit/Connect/OCKConnectTableViewCell.m]: @umerkhan-apple Sounds good. Were you thinking of clipping it like this?

Comment (umerkhan-apple) [inline CareKit/Connect/OCKConnectTableViewCell.m]: @stevemoser Same idea, but with a slightly different approach.
```
NSString *monogram = self.contact.monogram;
NSString *clippedMonogram = [monogram substringToIndex:MIN(monogram.length, 2)];
```

We should also update this in `OCKConnectTableViewHeader.m`

Comment (stevemoser) [inline CareKit/Connect/OCKConnectTableViewCell.m]: @umerkhan-apple Will do though I'll be gone for a few days so I will get to it when I get back. However while subStringToIndex is simpler it breaks surrogate pairs. Is the simplify worth it in this case? When I use subStringToIndex with my example above the emoji becomes a half 'A' in a box unicode character (sorry I don't know the exact name).

**Comment by matteocrippa** [comment]
Isn't better that monogram is auto-generated according the name provided by the user?
I added an on boarding page to provide physician details in my app and monogram is used to be generated with this string extension:

```
extension String {
    func getMonogram() -> String {
        var tmp = self.componentsSeparatedByString(" ")
        var name = ""
        var surname = ""
        name = String(tmp[0].characters.first!)
        if tmp.count > 1 {
            surname = String(tmp[1].characters.first!)
        }
        return name+surname
    }
}
```

**Comment by umerkhan-apple** [comment]
@matteocrippa I really like that idea! 
@stevemoser I agree that we don't have half of the unicode character in the monogram. How do you feel about auto-generating the monogram?

**Comment by oliverschaefer** [comment]
@matteocrippa @umerkhan-apple I like the idea of auto-generated monograms. When implementing, please keep in mind that the physician's name might contain (academic) titles. Some regions (e.g. Europe) place them in front of the given name. E.g. John Appleseed, M.D. (USA) = Dr. John Appleseed (Germany).

**Comment by matteocrippa** [comment]
@oliverschaefer that part it would be tricky, we probably have two way to approach that: blacklisting or check for dots inside the word. Let's talk about that in #21

**Comment by umerkhan-apple** [comment]
@stevemoser Let's proceed with this PR without adding the auto-generated monograms, as that will be added in #21.

**Comment by umerkhan-apple** [comment]
@stevemoser Could you resolve merge conflicts? PR looks good to be merged otherwise. 👍

**Comment by stevemoser** [comment]
@umerkhan-apple 
- [x] Updated OCKConnectTableViewHeader.m to adjust font size to fit
- [x] Move clipping logic to OCKContact
- [x] Updated header for OCKContact

Anything else?

File: CareKit/Connect/OCKContact.h [hunk 2]
```diff
@@ -138,6 +138,7 @@ OCK_CLASS_AVAILABLE
  A string indicating the monogram for a contact.
  
  If a monogram is not provided, the image will be used for the contact.
```
Comment (umerkhan-apple) [inline CareKit/Connect/OCKContact.h]: Remove this line.

File: CareKit/Connect/OCKContact.h [hunk 3]
```diff
@@ -138,6 +138,7 @@ OCK_CLASS_AVAILABLE
...
+ If a monogram is available, it will be clipped to two glyphs.
  */
```
Comment (umerkhan-apple) [inline CareKit/Connect/OCKContact.h]: I think it should just say "The monogram will be clipped to two glyphs."

**Comment by umerkhan-apple** [comment]
@stevemoser Minor nits. Apart from that, it is good to go! 👍

**Comment by stevemoser** [comment]
@umerkhan-apple And done. 👍

**Comment by umerkhan-apple** [comment]
@stevemoser Looks great! Merging.

**Pull request merged**